lib/customizer.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * Bootstraping the Gutenberg Customizer widget blocks section.
+ *
+ * Widget area edits made in the Customizer are synced to Customizer
+ * changesets as an object, encoded as a JSON string, where the keys
+ * are widget area IDs and the values are serialized block content.
+ * This file takes care of that syncing using the 2-way data binding
+ * supported by `WP_Customize_Control`s. The process is as follows:
+ *
+ * - On load, the client checks if the current changeset has
+ * widget areas that it can parse and use to hydrate the store.
+ * It will load all widget areas for the current theme, but if
+ * the changeset has content for a given area, it will replace
+ * its actual published content with the changeset's.
+ *
+ * - On edit, the client updates the 2-way bound input with a new object that maps
+ * widget area IDs and the values are serialized block content, encoded
+ * as a JSON string.
+ *
+ * - On publish, a PHP action will parse the JSON string in the
+ * changeset and update all the widget areas in it, to store the
+ * new content.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * The sanitization function for incoming values for the `gutenberg_widget_blocks` setting.
+ * It's a JSON string, so it decodes it and encodes it again to make sure it's valid.
+ *
+ * @param string $value The incoming value.
+ */
+function gutenberg_customize_sanitize( $value ) {
+	return json_encode( json_decode( $value ) );
+}
+
+/**
+ * Gutenberg's Customize Register.
+ *
+ * Adds a section to the Customizer for editing widgets with Gutenberg.
+ *
+ * @param \WP_Customize_Manager $wp_customize An instance of the class that controls most of the Theme Customization API for WordPress 3.4 and newer.
+ * @since 6.0.0
+ */
+function gutenberg_customize_register( $wp_customize ) {
+	require dirname( __FILE__ ) . '/class-wp-customize-widget-blocks-control.php';
+	$wp_customize->add_setting(
+		'gutenberg_widget_blocks',
+		array(
+			'default'           => '{}',
+			'type'              => 'gutenberg_widget_blocks',
+			'capability'        => 'edit_theme_options',
+			'transport'         => 'postMessage',
+			'sanitize_callback' => 'gutenberg_customize_sanitize',
+		)
+	);
+	$wp_customize->add_section(
+		'gutenberg_widget_blocks',
+		array( 'title' => __( 'Widget Blocks', 'gutenberg' ) )
```

Comment:
Would it make sense to name this section 'Widget Blocks (Experimental)', to make users more aware of the experimental nature of this section? Similar to what we did with the widget blocks screen and the legacy widgets block.

Reply:
Yes